design-documents/web-api-controllers.md

```diff
@@ -31,6 +31,28 @@ doesn't make sense in context of a customer managing itself since a customer can
 All of these problems stem from having business layer contracts/entities exposed as presentation layer operations/data.
  
  
+###### Examples of classes that contain presentation layer logic
+While service contracts implementations usually can contain all the logic needed to execute an operation various
+presentation areas would require additional logic to convey results to users or to extract operations' arguments from
+user input. Since service contracts are not supposed to know about context they are also not feet to contain additional
+authorization related logic.
+ 
+Examples of cases when we need additional context aware/presentation logic:
+* When we need to validate entity ownership before performing an operation using authenticated user entity
+* Limit frequency of operation execution (for anti-bruteforce measures)
```

Comment:
Why do we care about this on the application level instead of infrastructure level?

Reply:
we do have CAPTCHA in our application level since we want to protect our functionality and can't rely on merchants' infrastructure